liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java

```diff
@@ -361,16 +361,16 @@ public boolean recompileBuildFile(File buildFile, List<String> artifactPaths, Th
 
                         if (!dependencyIds.isEmpty()) {
                             runLibertyMavenPlugin("install-feature", serverName, dependencyIds);
-                            // update dependencies
-                            for (String dependencyId : dependencyIds) {
-                                for (Dependency dependency : dependencies) {
-                                    if (dependencyId.equals(dependency.getArtifactId())) {
-                                        log.info("found match: " + dependencyId);
-                                        this.existingDependencies.add(dependency);
-                                    }
+                            dependencyIds.clear();
+                        }
+
+                        // update dependencies
+                        for (Artifact artifact : updatedArtifacts) {
+                            for (Dependency dependency : dependencies) {
+                                if (artifact.getArtifactId().equals(dependency.getArtifactId())) {
+                                    this.existingDependencies.add(dependency);
```

Comment:
Can add a `break` to get out of the inner loop here, after the dependency is matched.